### ISSUE TITLE
`julia.withPackages`fails to build

### ISSUE THREAD

**Issue by fedeinthemix** [NixOS/nixpkgs#324654]
### Describe the bug
I'm trying to setup a `julia` environment with packages using `withPackages`, but get a build error.

Using the following flake.nix
```nix
  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
  inputs.flake-utils.url = "github:numtide/flake-utils";

  outputs = { self, nixpkgs, flake-utils }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        pkgs = nixpkgs.legacyPackages.${system};
        juliaPackages = [ "Plots"
                          #"GLMakie"
                          "CairoMakie" ];
      in {
        # nix develop
        devShells.default = pkgs.mkShell {
          packages = [
            (pkgs.julia.withPackages juliaPackages)
          ];
        };
      });
}

```
and running `nix develop` results in the error
```nix
$ nix develop 
warning: Git tree '/home/user/tmp/julia/test-1' is dirty
error: builder for '/nix/store/1mhgfxnpy129j4m7wmnfdz2rmvn7x9x7-DelaunayTriangulation.jl-26eb8e2.drv' failed with exit code 1;
       last 6 log lines:
       > exporting https://github.com/DanielVandH/DelaunayTriangulation.jl.git (rev 26eb8e2331b55735c3d305d949aabd7363f07ba7) into /nix/store/3kg5pc715rj2ymzd79ph36dimpzl4shc-DelaunayTriangulation.jl-26eb8e2
       > Initialized empty Git repository in /nix/store/3kg5pc715rj2ymzd79ph36dimpzl4shc-DelaunayTriangulation.jl-26eb8e2/.git/
       > fatal: could not read Username for 'https://github.com': No such device or address
       > fatal: could not read Username for 'https://github.com': No such device or address
       > fatal: could not read Username for 'https://github.com': No such device or address
       > Unable to checkout 26eb8e2331b55735c3d305d949aabd7363f07ba7 from https://github.com/DanielVandH/DelaunayTriangulation.jl.git.
       For full logs, run 'nix log /nix/store/1mhgfxnpy129j4m7wmnfdz2rmvn7x9x7-DelaunayTriangulation.jl-26eb8e2.drv'.
error: 1 dependencies of derivation '/nix/store/c25g3p8qmnvljia533yipa2n5q4y1s14-julia-DelaunayTriangulation-0.8.11.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6jgwxyb17rj956by3n757gxs024wqwg0-dependency-uuid-to-repo.yml.drv' failed to build
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/0z4i8wyg3ymm7kkb1gk2wyh08x230c6z-source/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'nix-shell'

         at /nix/store/0z4i8wyg3ymm7kkb1gk2wyh08x230c6z-source/pkgs/stdenv/generic/make-derivation.nix:375:7:

          374|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          375|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          376|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: 1 dependencies of derivation '/nix/store/qrim6xzsddzk13iyg4xp66jvvbhn5m7a-julia-artifacts.nix.drv' failed to build
```
### Steps To Reproduce
1. run `nix develop` on the above `flake.nix`

### Notify maintainers
@thomasjm @NickCao @joshniemela 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.05 (Uakari), 24.05.2355.d032c1a6dfad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - channels(user): `""`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by thomasjm** [comment]
This seems to be fixed now on master. I bisected and found it was fixed by this registry bump: eac352421d185289c24005e21a7d9ad360a55e4f. 

Seems like the failure was because the `DelaunayTriangulation.jl` repo was renamed from `https://github.com/DanielVandH/DelaunayTriangulation.jl.git` to `https://github.com/JuliaGeometry/DelaunayTriangulation.jl.git`. Slightly confused why this caused a failure, as I thought GitHub would provide a redirect from the old URL to the new one. Perhaps Nixpkgs's `fetchFromGitHub` is not following redirects?

Please let me know if you see any other problems.

**Comment by fedeinthemix** [comment]
Thanks. Can you please back-port the fix to `nixos-24.05`?

**Comment by thomasjm** [comment]
Done
